src/GoTrueClient.ts

```diff
@@ -499,8 +499,16 @@ export default class GoTrueClient {
     })
   }
 
-  private async _exchangeCodeForSession(authCode: string): Promise<AuthTokenResponse> {
-    const codeVerifier = await getItemAsync(this.storage, `${this.storageKey}-code-verifier`)
+  private async _exchangeCodeForSession(authCode: string): Promise<
+    | {
+        data: { session: Session; user: User; redirectType: string | null }
+        error: null
+      }
+    | { data: { session: null; user: null; redirectType: null }; error: AuthError }
+  > {
+    const [codeVerifier, redirectType] = (
+      (await getItemAsync(this.storage, `${this.storageKey}-code-verifier`)) as string
+    ).split('/')
```

Comment:
This throws `TypeError: Cannot read properties of null (reading 'split')` when the item is not in storage, e.g. if a magic link is opened in a different browser than used to create it.

Reply:
Thanks for notifying about this error. A PR to fix this issue is here:
https://github.com/supabase/gotrue-js/pull/825